server/container/guice/cassandra-rabbitmq-guice/src/test/java/org/apache/james/WithoutSwiftOrAwsS3Extension.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.james;
+
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+public class WithoutSwiftOrAwsS3Extension implements CassandraRabbitMQJamesServerTest, BeforeAllCallback, AfterAllCallback, BeforeEachCallback, AfterEachCallback, ParameterResolver {
+
+    JamesServerExtension jamesServerExtension;
+
+    WithoutSwiftOrAwsS3Extension() {
+        jamesServerExtension = baseExtensionBuilder().build();
+    }
+
+    @Override
+    public void beforeAll(ExtensionContext context) throws Exception {
```

Comment:
That's the 4th time I see this and I start to believe this is really `boiler plate`.

Why do we need 4 new extensions while previously we needed only one - that was generic - ?

Maybe we are trying to rely too much on parameter injection while we could rely on a getter and a method to be implemented and not need all these new boiler plate extensions.

Reply:
Antoine refactored that in order to have clearer view of where the tests failed